homeassistant/components/sensor/statistics.py

```diff
@@ -111,10 +110,22 @@ def async_stats_sensor_state_listener(entity, old_state, new_state):
 
             self._add_state_to_queue(new_state)
 
-            hass.async_add_job(self.async_update_ha_state, True)
+            self.hass.async_add_job(self.async_update_ha_state, True)
+
+        @callback
+        def async_stats_sensor_startup(event):
+            """Add listener and get recorded state."""
+            _LOGGER.debug("Startup for %s", self.entity_id)
+
+            async_track_state_change(
+                self.hass, self._entity_id, async_stats_sensor_state_listener)
+
+            if 'recorder' in self._hass.config.components:
+                # only use the database if it's configured
+                self.hass.async_add_job(self._initialize_from_database)
```

Comment:
Based on my understanding, because we're calling another method that is async (changed the name of this method now to _async_initialize_from_database) it should thus be async_create_task. If it would be a sync method then I would use async_add_executor_job correct?